src/Integration/Flysystem/S3/src/S3FilesystemV1.php

```diff
@@ -513,10 +513,20 @@ protected function getOptionsFromConfig(Config $config): array
      */
     protected function normalizeResponse(object $output, ?string $path = null): array
     {
+        if (empty($path)) {
+            if ($output instanceof AwsObject) {
+                $path = $output->getKey();
+            } elseif ($output instanceof CommonPrefix) {
+                $path = $output->getPrefix();
+            }
+            if (empty($path)) {
+                throw new \InvalidArgumentException('The provided path should not be null');
+            }
+
+            $path = $this->removePathPrefix($path);
+        }
         $result = [
-            'path' => $path ?: $this->removePathPrefix(
-                method_exists($output, 'getKey') ? $output->getKey() : (method_exists($output, 'getPrefix') ? $output->getPrefix() : null)
```

Comment:
Hey @Nyholm could you have a look to this? psalm complained about `this->removePathPrefix(possible-null);`

Reply:
Hm. This function supports quite a few different object types.. 

I'll have a look.

Reply:
It looks okey.